### ISSUE TITLE
[Components] ironclad

### ISSUE THREAD

**Issue by pipedream-component-development** [PipedreamHQ/pipedream#14837]
# ironclad
## URLs
- https://developer.ironcladapp.com/
#### Prompt
Emit new event when a fresh approval event is generated. Required props: selected event
#### URLs
- https://developer.ironcladapp.com/docs/getting-started
- https://developer.ironcladapp.com/reference/getting-started-api
- https://developer.ironcladapp.com/reference/authentication-api
- https://developer.ironcladapp.com/reference/pagination-api
- https://developer.ironcladapp.com/reference/workflows
- https://developer.ironcladapp.com/reference/records
- https://developer.ironcladapp.com/reference/webhooks
- https://developer.ironcladapp.com/reference/users
- https://developer.ironcladapp.com/reference/groups
- https://developer.ironcladapp.com/reference/schemas
- https://developer.ironcladapp.com/reference/authorization-1
- https://developer.ironcladapp.com/reference/resources-1
#### Prompt
Emit new event when a new workflow event is created. Required props: selected event
#### URLs
- https://developer.ironcladapp.com/reference/getting-started-api
- https://developer.ironcladapp.com/reference/authentication-api
- https://developer.ironcladapp.com/reference/pagination-api
- https://developer.ironcladapp.com/reference/workflows
- https://developer.ironcladapp.com/reference/records
- https://developer.ironcladapp.com/reference/webhooks
- https://developer.ironcladapp.com/reference/users
- https://developer.ironcladapp.com/reference/groups
- https://developer.ironcladapp.com/reference/schemas
- https://developer.ironcladapp.com/reference/authorization-1
- https://developer.ironcladapp.com/reference/resources-1
#### Prompt
Emit new event when a workflow document event is freshly established.
#### URLs
- https://developer.ironcladapp.com/docs/getting-started
- https://developer.ironcladapp.com/reference/workflows
- https://developer.ironcladapp.com/reference/webhooks
- https://developer.ironcladapp.com/reference/authentication-api
- https://developer.ironcladapp.com/reference/authorization-1
- https://developer.ironcladapp.com/reference/pagination-api
- https://developer.ironcladapp.com/reference/getting-started-api
#### Prompt
Launches a new workflow. Required props: workflow details. Optional props: attachments, user.
#### URLs
- https://developer.ironcladapp.com/docs/launch-a-workflow
- https://developer.ironcladapp.com/docs/bulk-launch-a-workflow
- https://developer.ironcladapp.com/reference/workflows
#### Prompt
Creates a new record in ironclad. Required props: record data. Optional props: user, tags.
#### URLs
- https://developer.ironcladapp.com/docs/create-a-record-1
- https://developer.ironcladapp.com/reference/workflows
- https://developer.ironcladapp.com/reference/records
- https://developer.ironcladapp.com/reference/users
- https://developer.ironcladapp.com/reference/groups
- https://developer.ironcladapp.com/reference/authorization-1
#### Prompt
Updates the metadata of an existing workflow. Required props: workflow ID, updated metadata.
#### URLs
- https://developer.ironcladapp.com/docs/launch-a-workflow
- https://developer.ironcladapp.com/docs/bulk-launch-a-workflow
- https://developer.ironcladapp.com/reference/workflows
- https://developer.ironcladapp.com/reference/records

**Comment by malexanderlim** [comment]
@michelle0927 I've added the scopes to the app, you should be able to reconnect now with the additional permissions!

**Comment by malexanderlim** [comment]
@vunguyenhung , can you try again? Just added additional permissions on Sergio's account.

**Comment by michelle0927** [comment]
@vunguyenhung I made some updates to improve error handling, prop descriptions, and parsing JSON in prop values. The workflow schema doesn't provide options for the properties. I had to go to the "Start a Workflow" form in Ironclad to see what the valid options are.

<img width="1121" alt="Screenshot 2024-12-10 at 2 31 39 PM" src="https://github.com/user-attachments/assets/0e25a726-b589-4481-8e1b-85f3a1ecd975">

<img width="903" alt="Screenshot 2024-12-10 at 2 32 06 PM" src="https://github.com/user-attachments/assets/8386931a-2a76-409e-a04c-a3bf724856ca">
